src/chronify/store.py

```diff
@@ -232,29 +355,125 @@ def load_table(self, path: Path, schema: TableSchema) -> None:
             msg = "Only Parquet files are currently supported: {path=}"
             raise NotImplementedError(msg)
 
-        self.create_view_from_parquet(schema.name, path)
+        self.create_view_from_parquet(schema, path)
         try:
             self._check_table_schema(schema)
             self._check_timestamps(schema)
         except InvalidTable:
-            with self._engine.begin() as conn:
-                conn.execute(text(f"DROP VIEW {schema.name}"))
+            self.drop_view(schema.name)
             raise
 
+    def delete_rows(self, name: str, time_array_id_columns: dict[str, Any]) -> None:
+        """Delete all rows matching the time_array_id_columns."""
+        table = self.get_table(name)
+        if table is None:
+            msg = f"No table with {name=} is stored"
+            raise TableNotStored(msg)
+
+        if not time_array_id_columns:
+            msg = "time_array_id_columns cannot be empty"
+            raise InvalidParameter(msg)
+
+        stmt = None
+        for column, value in time_array_id_columns.items():
+            if stmt is None:
+                stmt = delete(table).where(table.c[column] == value)
+            else:
+                stmt = stmt.where(table.c[column] == value)
+
+        assert stmt is not None
+        with self._engine.connect() as conn:
+            conn.execute(stmt)
+            conn.commit()
+
+        logger.info(
+            "Delete all rows from table {} with time_array_id_columns {}",
+            name,
+            time_array_id_columns,
+        )
+
+        stmt2 = select(table).limit(1)
+        is_empty = False
+        with self._engine.connect() as conn:
+            res = conn.execute(stmt2).fetchall()
+            if not res:
+                is_empty = True
+
+        if is_empty:
+            logger.info("Delete empty table {}", name)
+            self.drop_table(name)
+
+    def drop_table(self, name: str) -> None:
+        """Drop a table from the database."""
+        table = self.get_table(name)
+        if table is None:
+            msg = f"{name=}"
+            raise TableNotStored(msg)
+
+        self._metadata.drop_all(self._engine, tables=[table])
+        self._metadata.remove(table)
+        with self.engine.connect() as conn:
+            self._remove_schema(conn, name)
```

Comment:
Does this not require a `drop table {name}` execution in addition to schema?

Reply:
As I learned a little while ago in your PR, line 413 is incorrect. I interpreted the docstring to say that it will only drop the tables passed in the `tables` parameter. My first attempted solution to your problem called this method and all tables were dropped - the parameter wasn't respected. I need to change that line to call `DROP TABLE` instead.